substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -158,6 +161,15 @@ pub struct CreateCmd {
 	/// The chain type.
 	#[arg(value_enum, short = 't', default_value = "live")]
 	chain_type: ChainType,
+	/// If you're generating a config for a parachain
+	#[arg(long, value_enum, default_value = "false")]
+	parachain: bool,
+	/// The para ID for your chain.
+	#[arg(value_enum, short = 'p', default_value = "100")]
+	para_id: u32,
+	/// The relay chain you wish to connect to.
+	#[arg(value_enum, short = 'c', default_value = "polkadot")]
+	relay_chain: String,
```

Comment:
Why not put this into a completely different command?

Reply:
I agree. Maybe something like `make_parachain`?

Reply:
Yeah that makes more sense, @michalkucharczyk would it be a subcommand to `CreateCmd` (as we want all of those args anyways for a parachain config) or a completely separate command, similar to `GenesisBuildAction`?

Reply:
> This makes it easier to differentiate, otherwise, we will always have relay_chain and para_id present in all solo chain configs (set to null, i.e., "relay_chain": null).

Not sure if I understand this. I was thinking about scenario where `chain-spec-builder create` does not add any parachain related fields, and `create/make-parachain` simply injects them to chain-spec.

Sub-command is interesting idea, I like it. The only thing is that it is a little over-verbose on command line. We would need to always write `chain-spec-builder create solo ....` or `chain-spec-builder create para ....`, correct?

Reply:
We can use just plain json injection yeah.  I will try this approach.

> Sub-command is interesting idea, I like it. The only thing is that it is a little over-verbose on command line. We would need to always write chain-spec-builder create solo .... or chain-spec-builder create para ...., correct?

Yes, exactly, I am not sure how intuitive it is because of the reason you stated.  I will try your first approach because it seems simpler to the end user 👍 ty!

Reply:
I would say the overall strategy we have is to be `parachain-first`, as in, if a parameter is not specified, it should assume you are in a parachain context. With this, I suggest: 

```
chain-spec-builder create solo 
chain-spec-builder create para --relay-id x --para-id y
# should be an alias for para
chain-spec-builder create --relay-id x --para-id y
```

If with subcommands this is not possible, then I'd do the opposite: `chain-spec-builder create` builds a parachain, and `chain-spec-builder create solo` builds a solochain. 

I know this is more of a breaking change, but I think it is worthwhile. This tool is not wildly used now, but will be used a lot more in the omni-node drive future, so we better make it aligned with our standards while we can.

Reply:
`chain-spec-builder create` vs `chain-spec-builder create solo` is inconsistent IMO.
`chain-spec-builder create --solo` or `chain-spec-builder create --no-para` would look better.

Reply:
We also need defaults for `relay-id` and `para-id`.
If we don't want to hard-code defaults we should fail if those params are not given (and also `--solo` was not provided).

Reply:
I feel there is probably a way to do something "smarter" by looking at the cumulus pallets in the wasm runtime, to detect if it is a parachain or solochain

Reply:
A lot of `omni-*` tools can benefit from reading the runtime's metadata and doing checks and inference. That being said, I think the code to do it is not super trivial. Would make a great follow-up, but I would not block this PR for it, assuming the latest commits are something that majority approve.